### PR TITLE
saghmrossi/update

### PR DESCRIPTION
**Pull request by saghmrossi**
Update functionality for the wire protocol

### CODE REVIEW

File: src/client/wire_protocol/operations.rs [hunk 1]
```diff
@@ -36,13 +36,21 @@ pub enum Message {
         cursor_id: i64,
         starting_from: i32,
         number_returned: i32,
-        documents: Vec<BsonDocument>,
+        documents: Vec<bson::Document>,
+    },
+    OpUpdate {
+        header: Header,
+        // ZERO goes here
```
Comment (alabid) [inline src/client/wire_protocol/operations.rs]: update comment here

File: src/client/wire_protocol/operations.rs [hunk 2]
```diff
@@ -36,13 +36,21 @@ pub enum Message {
...
+        full_collection_name: String,
```
Comment (alabid) [inline src/client/wire_protocol/operations.rs]: update variable name

File: src/client/wire_protocol/operations.rs [hunk 3]
```diff
@@ -149,13 +192,13 @@ impl Message {
         let header = Header::with_query(total_length, request_id);
 
         Ok(Message::OpQuery { header: header, flags: flags,
-                              full_collection_name: full_collection_name,
+                              namespace: namespace,
```
Comment (vkarpov15) [inline src/client/wire_protocol/operations.rs]: General comment: these changes are outside the scope of this commit. It's easy to get carried away at the beginning of the project, but let's work on making changes more focused going forward.

File: tests/client/wire_protocol.rs [hunk 4]
```diff
@@ -210,3 +211,82 @@ fn insert_docs() {
...
+
+#[test]
+fn insert_update_then_query() {
+    match TcpStream::connect("localhost:27017") {
+        Ok(mut stream) => {
+            let mut doc = Document::new();
+            doc.insert("foo".to_owned(), FloatingPoint(42.0));
+            let docs = vec![doc];
+            let flags = OpInsertFlags::no_flags();
+            let name = "test.test".to_owned();
```
Comment (vkarpov15) [inline tests/client/wire_protocol.rs]: Separate collection names please.

File: src/client/wire_protocol/operations.rs (hunk 3 above)
Comment (saghmrossi) [inline src/client/wire_protocol/operations.rs]: Sorry, I'm  getting confused because I literally just got a comment on this pull request telling me to make these changes.
